### ISSUE TITLE
Issue with ClosureInverseMonoid

### ISSUE THREAD

**Issue by ChristopherRussell** [gap-packages/Semigroups#435]
4cef57f34d3ee0b3b624a9e101f84287a80e88b8 seems to have introduced a bug. I noticed an example where the `IdempotentGeneratedSubsemigroup` of a semigroup had an element that was not in the semigroup. @mtorpey helped me bisect the history to this commit but we were unsure how the changes were affecting this example.

```
gap> S := 
> InverseMonoid( [ PartialPerm( [ 1, 4 ], [ 1, 4 ] ), PartialPerm( [ 1, 3, 4, 5 ], [ 1, 6, 4, 2 ] ),
> PartialPerm( [ 1, 2, 4, 6 ], [ 1, 5, 4, 3 ] ), PartialPerm( [ 1, 4 ], [ 4, 1 ] ),
> PartialPerm( [ 1, 3, 4, 5 ], [ 1, 3, 4, 5 ] ), PartialPerm( [ 1, 2, 4, 6 ], [ 1, 2, 4, 6 ] ),
> PartialPerm( [ 1, 2, 3, 4, 5, 6, 7 ], [ 1, 2, 3, 4, 5, 6, 7 ] ) ] );
<inverse partial perm monoid of rank 7 with 7 generators>
gap> I := IdempotentGeneratedSubsemigroup(S);
<inverse partial perm monoid of rank 7 with 4 generators>
gap> I = Semigroup(Idempotents(S));
false
gap> Size(I); Size(Idempotents(S));
5
4
gap> Elements(S);
[ <identity partial perm on [ 1, 4 ]>, (1,4), <identity partial perm on [ 1, 3, 4, 5 ]>,
  [3,6][5,2](1)(4), <identity partial perm on [ 1, 2, 4, 6 ]>, [2,5][6,3](1)(4),
  <identity partial perm on [ 1, 2, 3, 4, 5, 6, 7 ]> ]
gap> Elements(I);
[ <identity partial perm on [ 1, 4 ]>, <identity partial perm on [ 1, 3, 4, 5 ]>,
  <identity partial perm on [ 1, 2, 4, 6 ]>, <identity partial perm on [ 1, 2, 3, 4, 5, 6 ]>,
  <identity partial perm on [ 1, 2, 3, 4, 5, 6, 7 ]> ]
```

**Comment by james-d-mitchell** [comment]
Thanks @ChristopherRussell and @mtorpey, I can't really investigate this this week, but will try next week. Unless you want to have a go at fixing it yourself @ChristopherRussell ?

**Comment by ChristopherRussell** [comment]
I have had a look.

`IdempotentGeneratedSubsemigroup(S)` calls the following:

```
InverseSemigroup(Idempotents( S ), rec(
    small := true,
    acting := false ));\
```
which results in the bug.

However
```InverseSemigroup(Idempotents( S ));```
returns the right answer

Does this help?

**Comment by ChristopherRussell** [comment]
And the bug seems to arise in `ClosureInverseMonoidOrSemigroupNC` where `S = One(coll)` and `coll` is the subsemigroup which InverseSemigroup has been called on. The line `  coll := Filtered(coll, x -> not x in S);` removes the identity element of `coll` and a potentially different element is added back in later.

```
  coll := Filtered(coll, x -> not x in S);
  if IsEmpty(coll) then
    return S;
  fi;

  # opts must be copied and processed before calling this function
  # coll must be copied before calling this function

  # Make coll closed under inverses
  coll := Set(coll);
  n    := Length(coll);
  for i in [1 .. n] do
    x := coll[i] ^ -1;
    if not x in coll then
      AddSet(coll, x);
    fi;
  od;

  if Constructor = InverseMonoid and not One(coll) in coll then
    AddSet(coll, One(coll));
  fi;

```

**Comment by james-d-mitchell** [comment]
Think I fixed this @ChristopherRussell can you please comment on the PR?

**Comment by wilfwilson** [comment]
This was resolved by PR #436.
